src/Microsoft.AspNetCore.SystemWebAdapters.SessionState/Serialization/JsonSessionKeySerializer.Shared.cs

```diff
@@ -0,0 +1,64 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Diagnostics.CodeAnalysis;
+using System.Text.Json;
+
+#if !NETFRAMEWORK
+using Microsoft.Extensions.Options;
+#endif
+
+namespace Microsoft.AspNetCore.SystemWebAdapters.SessionState.Serialization;
+
+internal class JsonSessionKeySerializer : ISessionKeySerializer
+{
+    private readonly JsonSessionSerializerOptions _options;
+
+#if NETFRAMEWORK
+    public JsonSessionKeySerializer(JsonSessionSerializerOptions options)
+    {
+        _options = options;
+    }
+#else
+    public JsonSessionKeySerializer(IOptions<JsonSessionSerializerOptions> options)
+    {
+        _options = options.Value;
+    }
+#endif
+
+    public bool TryDeserialize(string key, byte[] bytes, out object? obj)
+    {
+        if (_options.KnownKeys.TryGetValue(key, out var type))
+        {
+            obj = JsonSerializer.Deserialize(bytes, type);
+            return true;
+        }
+
+        if (_options.ThrowOnUnknownSessionKey)
+        {
+            throw new UnknownSessionKeyException(key);
+        }
```

Comment:
This option and exception seem out of place now that the method returns a bool. Methods using the Try pattern aren't supposed to throw. I'd expect the caller to check the return value and decide if it should throw or not.

Reply:
good point. I've moved that to the caller